src/algorithms.jl

```diff
@@ -0,0 +1,76 @@
+abstract type AbstractEnclosureAlgorithm end
+
+"""
+   NaturalEnclosure <: AbstractEnclosureAlgorithm
+
+Data type to compute the range of `f` over `X` using natural enclosure, that is simply
```

Comment:
concrete suggestion (should be applied to other cases)

```suggestion
Enclose the range of `f` over `X` using natural enclosure, that is simply
```

Reply:
we can revamp docs later

Reply:
Sorry i missed this. Definitely need to change `compute -> enclose` because we cannot compute the exact range but a (probably larger) enclosure. 

Whether or not keep the "data type" word can be nitpicked later